### PR TITLE
Have SaveDiffCal ensure that GroupingWorkspace's base class has a valid detector ID map

### PR DESCRIPTION
**Pull request by bolotovskyr**
**Description of work**
Currently `SpecialWorkspace2D`, the base class of `GroupingWorkspace`, builds its `detID_to_WI` map only if one of the explicit constructors is called.  If, however, `SpecialWorkspace2D` is built with a default constructor, the map is not built.  So depending on how `GroupingWorkspace` was created, the map might be null.  The purpose of this PR is to allow `SaveDiffCal` algorithm, which uses `GroupingWorkspace` as input, to ensure that the map _is_ built.
 
**To test:**
../mantid/build/bin/DataHandlingTest SaveDiffCalTest
../mantid/build/bin/DataObjectsTest SpecialWorkspace2DTest test_empty_detID_map

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by rboston628 — DISMISSED** [review]

**Comment by rboston628** [comment]
I forgot about documentation.  Please add release notes

**Review by rboston628 — DISMISSED** [review]

File: Framework/DataObjects/inc/MantidDataObjects/SpecialWorkspace2D.h [hunk 1]
```diff
@@ -47,6 +47,8 @@ class MANTID_DATAOBJECTS_DLL SpecialWorkspace2D : public Workspace2D {
   @return Standard string name  */
   const std::string id() const override { return "SpecialWorkspace2D"; }
 
+  bool IsDetectorIDMappingEmpty() const { return detID_to_WI.empty(); }
```
Comment (peterfpeterson) [inline Framework/DataObjects/inc/MantidDataObjects/SpecialWorkspace2D.h:50]: This one should change to pascalCase too

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Some function renaming please

File: Framework/DataObjects/inc/MantidDataObjects/SpecialWorkspace2D.h (hunk 1 above)
Comment (bolotovskyr) [reply Framework/DataObjects/inc/MantidDataObjects/SpecialWorkspace2D.h]: Done.

**Review by rboston628 — APPROVED** [review]

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**